backend/app/tests/anti_recommenders/test_anti_recommender_generator.py

```diff
@@ -0,0 +1,33 @@
+from pytest_mock import MockFixture
+
+from app.models.settings.settings import settings
+from app.models.anti_recommendation.anti_recommendation import AntiRecommendation
+from app.anti_recommenders.anti_recommender_generator import AntiRecommendationGenerator
+from app.anti_recommenders.open_ai.open_ai_normal_anti_recommender.open_ai_normal_anti_recommender import (
+    OpenAiNormalAntiRecommender,
+)
+
+
+def test_generate_anti_recommendations(
+    mocker: MockFixture,
+    anti_recommender_generator: AntiRecommendationGenerator,
+    record_key: str,
+    anti_recommendations: tuple[AntiRecommendation, ...],
+    model_response: str,
+) -> None:
+    """Test that AntiRecommenderGenerator.generate_anti_recommendations() yields anti-recommendations of a given record key."""
+
+    if settings.anti_recommender_type.lower() == "openai" and settings.openai_api_key:
```

Comment:
This should be test parameterization rather than depending on settings. You can skip tests / elide parameters if the API key is missing.

Reply:
Ideally, all environmental dependencies of the tests will be apparent in the test method parameters (i.e., fixtures) in pytest.